models/purchase.py

```diff
@@ -0,0 +1,41 @@
+from odoo import models, fields, api
+from odoo.exceptions import UserError
+from datetime import timedelta
+
+
+class PurchaseOrder(models.Model):
+    _inherit = 'purchase.order'
+
+    active = fields.Boolean('Is Active?', default=True)
+    us_phone_number = fields.Char('US Phone Number')
+
+    @api.model
+    def _check_archivability(self):
+        return len(self.filtered(lambda record: record.state not in ['done', 'cancel'])) > 0
+
+    @api.model
+    def _check_accessability(self):
+        return self.env.user.user_has_groups('purchase.group_purchase_manager')
```

Comment:
It would be better if we do like this:
```
@api.model
def _check_accessability(self):
      if not self.env.user.user_has_groups('purchase.group_purchase_manager'):
            raise UserError(
                'Only Administrators are allowed to archive records!')
      return True
```